spec/GraphQLOverHTTP.md

```diff
@@ -215,3 +215,31 @@ as long as the server is still able to produce a well-formed response.
 
 In case of errors that completely prevent the successful execution of the request,
 the server SHOULD respond with the appropriate status code depending on the concrete error condition.
+
+The following examples provide guidance on how to deal with specific error cases:
+
+### Unparseable or invalid request body
+
+For example: `NONSENSE`, `{"qeury": "{__typena`
+
+Completely prevents execution of the GraphQL operation and SHOULD result in status code `4ß0` (Bad Request).
+
+### Document validation
+
+Includes validation steps that run before execution of the GraphQL operation:
+- [GraphQL specification validation](http://spec.graphql.org/June2018/#sec-Validation)
+- custom validation, for example: depth limit, complexity limit
+
+The server MAY choose to:
+- deny execution, the status code SHOULD be `400` (Bad Request)
+- attempt execution, the status code will depend on the validity of the result
+
+### Runtime validation
+
+Validation steps performed by resolvers during execution of the GraphQL operation.
+The status code will depend on the validity of the result.
```

Comment:
In previous conversations, it was said that any validation done INSIDE the resolvers should result in a `2xx` (preferably a `200`) status code response.  Individual resolvers could have code called that is validations but are in reality arbitrary.

Reply:
That's a great point - but I'd rather stick to the idea of "any time a partial result is returned (non-null data) then it must be 2xx and not 4xx.  I don't think the spec should explicitly make requirements on the opaque parts of the server - if the client can't tell whether something happened inside or outside of resolvers then I don't think the spec should make assertions about it.  I'm not sure "non-null data" is enough, but maybe it's a starting point?

Reply:
Isn't it true that once any resolver executes, it guarantees a non-null data in the response?  If so, than the status code MUST be 2xx.

```suggestion
The status code (as mentioned above) MUST be 2xx, and SHOULD be 200 (Okay).


```

Reply:
It could be the case that a root resolves for a non-nullable field aborts because of a validation error. In that case `data` could be forced to become `null`, thus allowing (but not forcing) a non-2xx status code.

Since this section is of the spec is us giving recommendations for specific scenarios, I changed this to encourage returning `200` for consistency across fields: [`663ef69` (#125)](https://github.com/graphql/graphql-over-http/pull/125/commits/663ef69197e33c5625816b398366df65c7230837)